exchanges/stream/websocket_test.go

```diff
@@ -112,20 +198,50 @@ func TestIsDisconnectionError(t *testing.T) {
 	}
 
 	isADisconnectionError = isDisconnectionError(&net.OpError{
-		Op:     "",
-		Net:    "",
-		Source: nil,
-		Addr:   nil,
-		Err:    errors.New("errorText"),
+		Err: errClosedConnection,
+	})
+	if isADisconnectionError {
+		t.Error("It's not")
+	}
+
+	isADisconnectionError = isDisconnectionError(&net.OpError{
+		Err: errors.New("errText"),
 	})
 	if !isADisconnectionError {
 		t.Error("It is")
 	}
 }
 
 func TestConnectionMessageErrors(t *testing.T) {
+	var wsWrong = &Websocket{}
+	err := wsWrong.Connect()
+	if err == nil {
+		t.Fatal("error cannot be nil")
+	}
+
+	wsWrong.connector = func() error { return nil }
+	err = wsWrong.Connect()
+	if err == nil {
+		t.Fatal("error cannot be nil")
+	}
+
+	wsWrong.enabled = true
+	wsWrong.connecting = true
+	wsWrong.Wg = &sync.WaitGroup{}
+	err = wsWrong.Connect()
+	if err == nil {
+		t.Fatal("error cannot be nil")
+	}
+
+	wsWrong.connecting = false
```

Comment:
two more to use the protected funcs with

Reply:
Spotto